xcode/Subconscious/Shared/Components/Detail/MemoEditorDetail.swift

```diff
@@ -987,7 +987,7 @@ struct MemoEditorDetailModel: ModelProtocol {
         info: MemoEditorDetailDescription
     ) -> Update<MemoEditorDetailModel> {
         // No address? This is a draft.
-        guard let address = state.address ?? info.address else {
+        guard let address = info.address ?? state.address else {
```

Comment:
Reasoning about what this inversion is doing is tricky. Can you explain what's happening?

Reply:
When we rewrite the navigation stack after a rename we rewrite the list of `details` to point to the new address (with updated audience). When we pop the navigation stack we dispatch `send(.appear(description))` in `onAppear {}` which contains the _updated_ address. This is our chance to reload the data and refresh the view.

The bug was that we were throwing the _new_ address away (with updated audience) and retaining the address stored in `state`, which no longer resolves. 

TL;DR, in `.appear`'s handler, `state` is stale and `info` is fresh.

Reply:
That scans. Let's get this merged 🙏